chainer/dataset/tabular/from_data.py

```diff
@@ -0,0 +1,104 @@
+import chainer
+from chainer.dataset.tabular import tabular_dataset
+
+
+def from_data(*args, **kwargs):
```

Comment:
Is it possible that this can be made as a `class` method so we can call it as `TabularDataset.from_data`?

Reply:
Yes, it is an option.
One of my concern is that users can call `from_data` on an inheritance of `TabularDataset` but it does not return the class.
For example,
```python
class SomeDataset(TabularDataset):
    ...

# users may think a SomeDataset is constructed with data.
# but it is just a TabularDataset
dataset = SomeDataset.from_data(data)
```

Do you think this behaviour acceptable? 

I'm planning to add utilities under `tabular` name space so that users can import it.
```python
import chainer.dataset.tabular as td

dataset1 = td.from_data(data1)
dataset2 = td.from_data(data2)
dataset = td.concat(dataset1, dataset2)
```

Reply:
> Do you think this behaviour acceptable?

Nope. I see, I did not think about this. It just reminded me of the new `fastai` API.
Eg https://docs.fast.ai/tabular.data.html#TabularList.from_df

> I'm planning to add utilities under tabular name space so that users can import it.

My only concern was this does allow user to use `from_data` directly by importing it, which `classmethod` would have mitigated, but I didn't see the inheritance shortcoming.

Reply:
> My only concern was this does allow user to use from_data directly by importing it,

I feel this is an advantage. If a code uses a lot of `from_data`, user may prefer short form.